cerebrium/container-images/defining-container-images.mdx

```diff
@@ -0,0 +1,288 @@
+
+### Shell Commands
+
+Shell commands execute every time a new instance starts, making them ideal for runtime initialization and app startup:
+
+```toml
+shell_commands =  [
+    "echo 'Hello, World!'",
+    "uv venv",
+    "export SOME_ENV_VAR=value"
+]
+```
+
+Shell commands are perfect for tasks that need fresh execution on startup, such as downloading models, warming caches, or starting an app server. Since these commands run on every instance start, they should be optimized for speed to minimize startup time.
+
+### Pre-build Commands
+
+Pre-build commands execute once during image creation, making them suitable for setup tasks that can be baked into the container (Which should also improve cold start performance):
+
+```toml
+pre_build_commands = [
+    "curl -o /usr/local/bin/pget -L 'https://github.com/replicate/pget/releases/download/v0.6.2/pget_linux_x86_64' && chmod +x /usr/local/bin/pget"
+]
+```
+
+Pre-build commands are ideal for compiling code, building assets, or installing build-time dependencies. Since these commands affect the container image, changes to pre-build commands trigger a full rebuild. Group related commands together to minimize rebuild frequency.
+
+## Custom Docker Base Images
+
+The choice of base image fundamentally shapes how an app runs in Cerebrium. While the default Debian slim image works well for most Python apps, some scenarios require specialized base images.
+
+### Selecting Base Images
+
+The base image selection shapes how an app runs in Cerebrium. While the default Debian slim image works for most Python apps, other validated base images support specific requirements.
+
+### Supported Base Images
+
+Cerebrium supports several categories of base images to ensure system compatibility:
+
+```toml
+# Default minimal image
+docker_base_image_url = "debian:bookworm-slim"
+
+# CUDA-enabled images
+docker_base_image_url = "nvidia/cuda:12.0.1-runtime-ubuntu22.04"
+```
+
+The system accepts these image types:
+
+#### Ubuntu-based CUDA Images
+
+All `nvidia/cuda` images that include Ubuntu are supported. These provide GPU acceleration capabilities:
+
+```toml
+docker_base_image_url = "nvidia/cuda:12.0.1-runtime-ubuntu22.04"
+docker_base_image_url = "nvidia/cuda:12.0.1-devel-ubuntu22.04"
+```
+
+#### Debian and Ubuntu Base Images
+
+Any Debian or Ubuntu base image works as a foundation:
+
+```toml
+docker_base_image_url = "debian:bullseye"
+docker_base_image_url = "ubuntu:22.04"
+```
+
+#### Python Images
+
+Python images based on Debian bullseye or bookworm provide pre-configured Python environments:
+
+```toml
+docker_base_image_url = "python:3.11-bullseye"
+docker_base_image_url = "python:3.11-bookworm"
+```
+
+<Tip>
+  It is better to start with a debian or ubuntu image and then add more
+  complexity as development proceeds. `nvidia/cuda` images are bloated and
+  affect both cold start and build times.
+</Tip>
```

Comment:
I think good to add that container image size and startup times are linked - even if not linearly